### PR TITLE
[libc] added quick_exit function

### PR DESCRIPTION
**Pull request by RoseZhang03**
- In /libc/src/__support/ OSUtil, changed quick_exit to just exit, and put in namespace
LIBC_NAMESPACE::internal.
- In /libc/src/stdlib added quick_exit
- Added test files for quick_exit

### CODE REVIEW

**Review by aaryanshukla — COMMENTED** [review]
Just some marginal formatting issues like maximum amount of text on a line and some of the spacing between lines. 

libc/src/stdlib/quick_exit.cpp: adjust the code format on the "Quick Implementation Exit" goes beyond 80. 

libc/src/stdlib/quick_exit.h: Adjust the block comment format since it goes beyond 80.

File: libc/src/stdlib/quick_exit.cpp
Comment (aaryanshukla) [inline libc/src/stdlib/quick_exit.cpp:1]: Line 1 exceeds 80 text per line.

File: libc/src/stdlib/quick_exit.h
Comment (aaryanshukla) [inline libc/src/stdlib/quick_exit.h:1]: Line 1 exceeds 80 text per the line.

**Review by michaelrj-google — COMMENTED** [review]
Here's what I found in review:

**Comment by amykhuang** [comment]
super nit: for the commit message I'd summarize the change first (a line, or a paragraph, or whatever) and then you can add the bullet points below. Also they're usually written in present tense for some reason.

File: libc/src/stdlib/CMakeLists.txt [hunk 1]
```diff
@@ -42,6 +42,17 @@ add_entrypoint_object(
...
 )
 
+add_entrypoint_object(
+  quick_exit
+  SRCS
+    quick_exit.cpp
+  HDRS
+    quick_exit.h
+  DEPENDS
+    libc.src.errno.errno
```
Comment (lntue) [inline libc/src/stdlib/CMakeLists.txt:52]: Does not look like `errno` is used in the quick_exit implementation?

File: libc/src/stdlib/quick_exit.h [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef LLVM_LIBC_SRC___SUPPORT_OSUTIL_QUICK_EXIT_H
-#define LLVM_LIBC_SRC___SUPPORT_OSUTIL_QUICK_EXIT_H
+#ifndef LLVM_LIBC_SRC_STDLIB_QUICK_EXIT_H
+#define LLVM_LIBC_SRC_STDLIB_QUICK_EXIT_H
+
+#include <stdlib.h>
```
Comment (lntue) [inline libc/src/stdlib/quick_exit.h:12]: is this `#include` needed?

File: libc/test/src/stdlib/quick_exit_test.cpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+//===----------------------------------------------------------------------===//
+
+#include "src/stdlib/exit.h"
+#include "src/stdlib/quick_exit.h"
+#include "test/UnitTest/Test.h"
+
+TEST(LlvmLibcStdlib, quick_exit) {
+  EXPECT_EXITS([] { LIBC_NAMESPACE::quick_exit(1); }, 1);
+  EXPECT_EXITS([] { LIBC_NAMESPACE::quick_exit(65); }, 65);
+}
```
Comment (lntue) [inline libc/test/src/stdlib/quick_exit_test.cpp:16]: Missing extra line at the end of file.

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Pull request merged**